pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -199,11 +200,9 @@ func (c *Reconciler) ReconcileKind(ctx context.Context, pr *v1beta1.PipelineRun)
 		before = pr.Status.GetCondition(apis.ConditionSucceeded)
 	}
 
-	vp, err := getVerificationPolicies(ctx, c.verificationPolicyLister, pr.Namespace)
-	if err != nil {
-		return fmt.Errorf("failed to list VerificationPolicies from namespace %s with error %v", pr.Namespace, err)
-	}
-	getPipelineFunc := resources.GetVerifiedPipelineFunc(ctx, c.KubeClientSet, c.PipelineClientSet, c.resolutionRequester, pr, vp)
+	// we don't have a feature flag to directly gate this code, in case it fail the pipelinerun we omit the error and defer it to trusted resources to check
+	verificationpolicies, _ := c.verificationPolicyLister.VerificationPolicies(pr.Namespace).List(labels.Everything())
```

Comment:
I'm thinking about a case that the lister is not able to list the policies (thought it could be extremely rare). If this case happens, it will stop users who even don't enable trusted resources. So I think maybe leave it to "trusted resources" code to check if policies exist.

Reply:
Wasn't this previously also the case though? In general, we want all errors to be handled (https://google.github.io/styleguide/go/decisions#handle-errors). If the controller returns an error, knative will add the taskrun/pipelinerun back onto the work queue to retry, unless it's a permanent error. So I think handling the error is the right decision here, as long as we can make sure a lister failure doesn't hit this case where a permanent error is returned: https://github.com/tektoncd/pipeline/blob/6c2d01c272c06caf952ad6ce182abf717da53afb/pkg/reconciler/taskrun/taskrun.go#L352-L355

Reply:
I think the difference is that, if we handle this failure here, this would be a possible failure to stable version users. Though this is a alpha feature. Previously this is gated by our feature flag. 

But I'm convinced by your comment. 😄 I will add it

Reply:
oh I see- makes sense. I agree we wouldn't want to return a permanent error due to the lister failure for users who don't care about trusted resources. There are two ways around this I could see: 1. create another error type for transient errors or 2. refactor error handling so a permanent error is returned only for things that are actually permanent. I think 2 might be a bit better but I haven't tried to write it out, so let me know if this suggestion is off base.

Reply:
Am I correct that if we are not returning with `controller.NewPermanentError(err)` then the error is not permanent error?

Reply:
Ok I created a new error and also add test to make sure that it is not a PermanentError. Then wrap the lister error into it to return

Reply:
Resolved offline-- I misunderstood how `GetVerifiedPipelineFunc` was called